bokehjs/test/integration/plots.ts

```diff
@@ -107,6 +107,112 @@ describe("Plot", () => {
     })
   })
 
+  describe("should support windowed auto-ranging", async () => {
+    function plot(window_axis: WindowAxis) {
+      const r = new Range1d({start: 0, end: 6})
+      const dr = new DataRange1d({range_padding: 0})
+      const p = figure({
+        sizing_mode: "fixed",
+        width: 300,
+        height: 300,
+        window_axis,
+        title: `window_axis == ${window_axis}`,
+        x_range: window_axis == "x" ? r: dr,
+        y_range: window_axis == "x" ? dr: r,
+      })
+      const s = p.scatter({x: [1, 2, 3, 4, 5], y: [1, 2, 3, 4, 5], size: 8})
+      return {p, s, r, dr}
+    }
+    for (const wax of ["x", "y"]) {
+      it(`with window_axis='${wax}' when data changes`, async () => {
+        const nwax = (wax == "x") ? "y" : "x"
+        const {p, s, dr} = plot(wax as WindowAxis)
+        const {view} = await display(p)
+
+        expect(dr.start).to.be.equal(1)
+        expect(dr.end).to.be.equal(5)
+
+        // updated data in fixed range
+        const data: any = {}
```

Comment:
Any more specific type is better than literal `any`, e.g. `{[key: string]: number[]}` or even better:
```suggestion
        import type {Data} from "@bokehjs/core/types"
        const data: Data = {}
```
Some in other cases.

Reply:
https://github.com/bokeh/bokeh/pull/14353/commits/bb9f114897c1b9fc9dae637fb8460d386c189179